packages/panzer/disc-fe/src/Panzer_FaceToElement_impl.hpp

```diff
@@ -95,21 +97,18 @@ initialize(panzer::ConnManager & conn,
   int nprocs = comm->getSize();
 #endif
 
+  if ( dimension == 1 ) {
+    panzer::EdgeFieldPattern edge_pattern(ebt[0]);
+    conn.buildConnectivity(edge_pattern);
+  } else if ( dimension == 2 ){
+    panzer::FaceFieldPattern face_pattern(ebt[0]);
+    conn.buildConnectivity(face_pattern);
+  } else {
+    panzer::ElemFieldPattern elem_pattern(ebt[0]);
+    conn.buildConnectivity(elem_pattern);
+  }
```

Comment:
Does this mean that we now require that the shards type has to be identical for all blocks? Does this need to be checked via an assert?

Reply:
That has always been the case. All element blocks are required to have the same topology right now. DOF manager and all assembly code have always made that assumption. This code was written assuming we would support hybrid meshing soon and did some inefficient things in anticipation of this capability.